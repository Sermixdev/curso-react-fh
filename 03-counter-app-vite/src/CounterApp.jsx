import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({ title, value }) => {

    const [counter, setCounter] = useState(value);

    const handleAdd = () => setCounter(counter + 1);
    const handleDel = () => setCounter(counter - 1);
    const handleRes = () => setCounter(value);

    return (
        <>
            <h1>{title}</h1>
            <h2>{counter}</h2>
            <button onClick={handleAdd}>+1</button>
            <button onClick={handleDel}>-1</button>
            <button onClick={handleRes}>Reset</button>
        </>
    );
};

CounterApp.propTypes = {
    title: PropTypes.string.isRequired,
    value: PropTypes.number.isRequired,
};
CounterApp.defaultProps = {
    title: 'No hay titulo',
    value: 0,
};
