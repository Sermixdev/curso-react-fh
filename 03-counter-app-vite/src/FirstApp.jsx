// export const FirstApp = () => <h2>Sergio</h2>;

// Retornar Fragmento de componente
const getResult = () => {
    return 4 + 4;
};
export const FirstApp = () => {
    return (
        <>
            <h1>{getResult()}</h1>
            {/* 
            <code>{JSON.stringify(newMessage)}</code> 
            */}
            <h2>Soy un subt√≠tulo</h2>
        </>
    );
};
