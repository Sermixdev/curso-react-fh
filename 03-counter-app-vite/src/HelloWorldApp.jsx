// export function App() {
//     return <h1>Hola mundo!!!!!!</h1>;
// }

export const HelloWorldApp = () => {
    return <h1>Hello World App</h1>;
};

// Lo mismo mÃ©todo corto
// export const HelloWorldApp = () => <h1>Hola mundo</h1>;
