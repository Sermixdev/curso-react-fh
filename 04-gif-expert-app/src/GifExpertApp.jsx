import { useState } from 'react';
import { AddCategory } from './components/AddCategory';
export const GifExpertApp = () => {
    const [categories, setCategories] = useState(['One Punch', 'Loco']);
    const onAddCategory = () => {
        // categories.push('Valorant) ESTA FORMA NO ES CORRECTA PORQUE PROVOCA MUTACION
        setCategories(['Valorant', ...categories]);
        // setCategories(cat => [...cat, 'Valorant'])
    };
    return (
        <>
            {/* titulo */}
            <h1>GifExpertApp</h1>
            {/* Input */}
            <AddCategory setCategories={setCategories} />
            {/* Listado de gif */}
            <ol>
                {categories.map((category) => {
                    return <li key={category}>{category}</li>;
                })}
            </ol>
            {/* Gif Item */}
        </>
    );
};
