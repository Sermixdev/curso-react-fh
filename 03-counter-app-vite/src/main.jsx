import React from 'react';
import ReactDOM from 'react-dom/client';

import { FirstApp } from './FirstApp';
// import { HelloWorldApp } from './HelloWorldApp';
// import { CounterApp } from './CounterApp';
import './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        {/* <CounterApp title='CounterApp' value={0} /> */}
        <FirstApp title="Hola, soy GOKU"></FirstApp>
    </React.StrictMode>
);
