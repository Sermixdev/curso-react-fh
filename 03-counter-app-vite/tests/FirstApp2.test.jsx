import { render, screen } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';

describe('Pruebas en FirstApp', () => {
    const title = 'hola soy goku';
    const subTitle = 'Hola soy un subtitulo';
    test('should match with snapshot', () => {
        const { container } = render(<FirstApp title={title} />);
        expect(container).toMatchSnapshot();
    });

    test('debe de mostrar el mensaje de "Hola, soy goku"', () => {
        // screen.debug();
        render(<FirstApp title={title} />);
        expect(screen.getByText(title)).toBeTruthy();
    });

    test('should show the title in h1', () => {
        render(<FirstApp title={title} />);
        expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title);
    });

    test('debe mostrar el subtitulo enviado por props', () => {
        render(<FirstApp title={title} subTitle={subTitle} />);
        expect(screen.getAllByText(subTitle).length).toBe(1)
    });
});
