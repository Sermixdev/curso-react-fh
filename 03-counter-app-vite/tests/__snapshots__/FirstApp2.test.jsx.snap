// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp should match with snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    hola soy goku
  </h1>
  <p>
    No hay subt√≠tulo
  </p>
  <p>
    Sergio Ferrer
  </p>
</div>
`;
